src/binder/assemblybinder.cpp

```diff
@@ -685,19 +685,23 @@ namespace BINDER_SPACE
 
         StackSString sCoreLib;
 
-        // At run-time, System.Private.CoreLib.ni.dll is typically always available, and
-        // System.Private.CoreLib.dll is typically not.  So check for the NI first.
+        // At run-time, System.Private.CoreLib.dll is expected to be the NI image.
+        // Try binding to it first as that. If it fails, we will try to bind to it as 
+        // IL image.
         sCoreLib = sCoreLibDir;
-        sCoreLib.Append(CoreLibName_NI_W);
+        sCoreLib.Append(CoreLibName_IL_W);
+#if defined(FEATURE_PAL) && !defined(_TARGET_AMD64_)
+        // Native image for core library only exists for X64 on non-Windows. As other architectures
```

Comment:
> the resulting bind still have the precompiled native code kick-in 

There should be nothing fundamental preventing it. There may be some other random check that derails it that we need to remove.